modules/k8s/tunnel.go

```diff
@@ -66,20 +66,34 @@ type Tunnel struct {
 	kubectlOptions *KubectlOptions
 	resourceType   KubeResourceType
 	resourceName   string
+	logger         logger.TestLogger
 	stopChan       chan struct{}
 	readyChan      chan struct{}
 }
 
-// NewTunnel will create a new Tunnel struct. Note that if you use 0 for the local port, an open port on the host system
-// will be selected automatically, and the Tunnel struct will be updated with the selected port.
+// NewTunnel creates a new tunnel with NewTunnelWithLogger, setting logger.Terratest as the logger.
 func NewTunnel(kubectlOptions *KubectlOptions, resourceType KubeResourceType, resourceName string, local int, remote int) *Tunnel {
+	return NewTunnelWithLogger(kubectlOptions, resourceType, resourceName, local, remote, logger.Terratest)
+}
+
+// NewTunnelWithLogger will create a new Tunnel struct with the provided logger.
+// Note that if you use 0 for the local port, an open port on the host system
+// will be selected automatically, and the Tunnel struct will be updated with the selected port.
+func NewTunnelWithLogger(
+	kubectlOptions *KubectlOptions,
+	resourceType KubeResourceType,
+	resourceName string,
+	local int,
+	remote int,
+	logger logger.TestLogger) *Tunnel {
```

Comment:
NIT: The style we use is:
```suggestion
	logger logger.TestLogger,
) *Tunnel {
```

Reply:
makes sense!